File: backend/src/main/java/com/bths/platform/quarto/Quarto.java
package com.bths.platform.quarto;

import com.bths.platform.quarto.enums.StatusQuarto;
import com.bths.platform.quarto.enums.TipoQuarto;
import com.bths.platform.viagem.Viagem;
import jakarta.persistence.*;

@Entity
@Table(name = "quartos")
public class Quarto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoQuarto tipo;


    @Column(nullable = false)
    private Integer capacidade;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusQuarto status;

    @ManyToOne
    @JoinColumn(name = "viagem_id", nullable = false)
    private Viagem viagem;

    public Quarto() {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public TipoQuarto getTipo() {
        return tipo;
    }

    public void setTipo(TipoQuarto tipo) {
        this.tipo = tipo;
    }

    public Integer getCapacidade() {
        return capacidade;
    }

    public void setCapacidade(Integer capacidade) {
        this.capacidade = capacidade;
    }

    public StatusQuarto getStatus() {
        return status;
    }

    public void setStatus(StatusQuarto status) {
        this.status = status;
    }

    public Viagem getViagem() {
        return viagem;
    }

    public void setViagem(Viagem viagem) {
        this.viagem = viagem;
    }
}

File: frontend/src/features/quarto/types/quarto.ts
export type TipoQuarto =
  | "SUITE"
  | "ALOJAMENTO";

export type StatusQuarto =
  | "DISPONIVEL"
  | "INDISPONIVEL";

export type Quarto = {
  id: number;
  nome: string;
  tipo: TipoQuarto;
  capacidade: number;
  status: StatusQuarto;

  viagemId: number;
  viagemNome: string;
};

export type QuartoOcupacao = {
  quartoId: number;
  nome: string;
  tipo: TipoQuarto;
  status: StatusQuarto;
  capacidade: number;
  ocupacao: number;
  vagasDisponiveis: number;
};